v-next/hardhat/test/internal/builtin-plugins/gas-analytics/gas-analytics-manager.ts

```diff
@@ -0,0 +1,1001 @@
+import type { GasMeasurement } from "../../../../src/internal/builtin-plugins/gas-analytics/types.js";
+
+import assert from "node:assert/strict";
+import path from "node:path";
+import { afterEach, describe, it } from "node:test";
+
+import {
+  getAllFilesMatching,
+  readJsonFile,
+  remove,
+} from "@nomicfoundation/hardhat-utils/fs";
+import chalk from "chalk";
+
+import {
+  avg,
+  median,
+  getUserFqn,
+  getFunctionName,
+  findDuplicates,
+  roundTo,
+  GasAnalyticsManagerImplementation,
+} from "../../../../src/internal/builtin-plugins/gas-analytics/gas-analytics-manager.js";
+
+describe("gas-analytics-manager", () => {
+  describe("GasAnalyticsManager", () => {
+    describe("constructor", () => {
+      it("should initialize with empty gasMeasurements array", () => {
+        const manager = new GasAnalyticsManagerImplementation(process.cwd());
+        assert.deepEqual(manager.gasMeasurements, []);
+      });
+    });
+
+    describe("addGasMeasurement", () => {
+      it("should add function gas measurement to the array", () => {
+        const manager = new GasAnalyticsManagerImplementation(process.cwd());
+        const functionMeasurement: GasMeasurement = {
+          type: "function",
+          contractFqn: "project/contracts/MyContract.sol:MyContract",
+          functionSig: "transfer(address,uint256)",
+          gas: 25000,
+        };
+
+        manager.addGasMeasurement(functionMeasurement);
+
+        assert.equal(manager.gasMeasurements.length, 1);
+        assert.deepEqual(manager.gasMeasurements[0], functionMeasurement);
+      });
+
+      it("should add deployment gas measurement to the array", () => {
+        const manager = new GasAnalyticsManagerImplementation(process.cwd());
+        const deploymentMeasurement: GasMeasurement = {
+          type: "deployment",
+          contractFqn: "project/contracts/MyContract.sol:MyContract",
+          gas: 500000,
+          size: 2048,
+        };
+
+        manager.addGasMeasurement(deploymentMeasurement);
+
+        assert.equal(manager.gasMeasurements.length, 1);
+        assert.deepEqual(manager.gasMeasurements[0], deploymentMeasurement);
+      });
+    });
+
+    describe("saveGasMeasurements", () => {
+      afterEach(async () => {
+        await remove(path.join(process.cwd(), "gas-stats"));
+      });
+
+      it("should save gas measurements in memory", async () => {
+        const manager = new GasAnalyticsManagerImplementation(process.cwd());
```

Comment:
I think you should use a tmp dir for this. Otherwise, if any of these tests fail after a measurement has been written, you'll have to manually delete the gas-stats directory wherever you run the tests.

With that approach you can also drop the `afterEach` cleanup.